actors/builtin/miner/miner_state.go

```diff
@@ -957,6 +985,133 @@ func (st *State) checkPrecommitExpiry(store adt.Store, sectors abi.BitField) (de
 	return depositToBurn, nil
 }
 
+type AdvanceDeadlineResult struct {
+	PledgeDelta                                abi.TokenAmount
+	PowerDelta                                 PowerPair
+	UndeclaredFaultyPower, DeclaredFaultyPower PowerPair
```

Comment:
It took me quite a while to figure out how (I think) these are being used. I think:
- UndeclaredFaultyPower is the faults triggered by missing posts
- DeclaredFaultyPower was not just declared here, but is the ongoing faulty power to be penalised

I.e. one is an "edge" and one is a "gauge", just reading a value from state. "Declared" is misleading here, since nothing is being declared and also it includes power from faults that were both previously, and in this call, undeclared.

Maybe names could help, but it definitely needs explanation. Maybe
- `DetectedFaultyPower` sum of power of sectors from partitions which were not proven this deadline
- `OngoingFaultyPower` sum of power of sectors that are (or were) faulty

Does the latter include the former? Maybe `PreviouslyFaultyPower` or `AlreadyFaultyPower` then.